contexts/permit-context.tsx

```diff
@@ -0,0 +1,122 @@
+import React, { useCallback } from 'react'
+import { usePublicClient, useWalletClient } from 'wagmi'
+import { hexToSignature } from 'viem'
+
+import { Currency } from '../model/currency'
+import { zeroBytes32 } from '../utils/bytes'
+import { TOKEN_VERSIONS } from '../utils/token-versions'
+
+import { useCurrencyContext } from './currency-context'
+
+type PermitContext = {
+  permit: (
+    asset: Currency,
+    owner: `0x${string}`,
+    spender: `0x${string}`,
+    value: bigint,
+    deadline: bigint,
+  ) => Promise<{
+    r: `0x${string}`
+    s: `0x${string}`
+    v: number
+  }>
+}
+
+const Context = React.createContext<PermitContext>({
+  permit: () =>
+    Promise.resolve({
+      r: zeroBytes32 as `0x${string}`,
+      s: zeroBytes32 as `0x${string}`,
+      v: 0,
+    }),
+})
+
+export const PermitProvider = ({ children }: React.PropsWithChildren<{}>) => {
```

Comment:
I think `permit` could just be a util function. Using context for this is overkill.